test/unittests/UserTest.php

```diff
@@ -0,0 +1,526 @@
+                                         'CenterID' => '1'),
+                              1 => array('UserID' => '1',
+                                         'CenterID' => '4')
+                        );
+    /**
+     * Examiners_psc_rel table information
+     *
+     * @var array
+     */
+    private $_eprInfo = array(0 => array('centerID' => '1',
+                                         'examinerID' => 1,
+                                         'active' => 'Y',
+                                         'pending_approval' => 'N'),
+                              1 => array('centerID' => '4',
+                                         'examinerID' => 1,
+                                         'active' => 'Y',
+                                         'pending_approval' => 'N')
+                        );
+    /**
+     * User object used for testing
+     *
+     * @var User object
+     */
+    private $_user;
+    /**
+     * NDB_Factory used in tests.
+     * Test doubles are injected to the factory object.
+     *
+     * @var NDB_Factory
+     */
+    private $_factory;
+    /**
+     * Test double for NDB_Config object
+     *
+     * @var \NDB_Config | PHPUnit_Framework_MockObject_MockObject
+     */
+    private $_configMock;
+    /**
+     * Test double for Database object
+     *
+     * @var \Database | PHPUnit_Framework_MockObject_MockObject
+     */
+    private $_dbMock;
+    /**
+     * Maps config names to values
+     * Used to set behavior of NDB_Config test double
+     *
+     * @var array config name => value
+     */
+    private $_configMap = array();
+    /**
+     * This method is called before each test is executed.
+     *
+     * @return void
+     */
+    protected function setUp(): void
+    {
+        parent::setUp();
+        $this->_factory = NDB_Factory::singleton();
+        $this->_factory->reset();
+        $this->_factory->setTesting(false);
+        $this->_configMock = $this->_factory->Config(CONFIG_XML);
+        $database     = $this->_configMock->getSetting('database');
+        $this->_dbMock     = Database::singleton(
+            $database['database'],
+            $database['username'],
+            $database['password'],
+            $database['host'],
+            1
+        );
+
+        $this->_username = "968775";
+    }
+
+    /**
+     * Tears down the fixture, for example, close a network connection.
+     * This method is called after a test is executed.
+     *
+     * @return void
+     */
+    protected function tearDown()
+    {
+        parent::tearDown();
+        $this->_factory->reset();
+    }
+
+    /**
+     * Test factory() method retrieves all _candidate and related info
+     *
+     * @return void
+     * @covers User::singleton
+     * @covers User::factory
+     * @covers User::getData
+     */
+    public function testFactoryRetrievesUserInfo()
+    {
+        $this->_setUpTestDoublesForFactoryUser();
+        $this->_user = \User::factory($this->_username);
```

Comment:
Is it possible to do this once in `setUp()` instead of including it in every unit test?

Reply:
I don't think so because the factory() method essentially just fills the user object with information from the database. If you take a look at the [Candidate unit tests](https://github.com/aces/Loris/blob/minor/test/unittests/CandidateTest.php) (including the ones written not by me), it's similar to how the select() method is used in every unit test to populate the Candidate object with information.